client/my-sites/plugins/plugin-details-CTA/index.jsx

```diff
@@ -465,6 +465,23 @@ function LegacyPluginDetailsCTA( {
 	);
 }
 
+function useSiteHasFeature( selectedSite, pluginFeature ) {
+	const shouldUpgrade = useSelector(
+		( state ) => ! siteHasFeature( state, selectedSite?.ID, pluginFeature )
+	);
+	const { product_slug } = selectedSite?.plan || {};
+
+	if (
+		! config.isEnabled( 'marketplace-personal-premium' ) &&
+		! shouldUpgrade &&
```

Comment:
I think this one needs to be removed? It shouldn't matter what the outcome is of `shouldUpgrade`, if the current site has a personal or premium plan it needs to upgrade.

So `config.isEnabled( 'marketplace-personal-premium' ) && ( isPersonalPlan( product_slug ) || isPremiumPlan( product_slug ) )`?

Reply:
Yeah, good point, if it was `true` it would be returned later down the line anyway. But I agree with the simpler version you suggested.

Reply:
Update at https://github.com/Automattic/wp-calypso/pull/66362/commits/0c4d1b924831af526738e75710d8fe8c24492cc7